### ISSUE TITLE
Adding a new milestone does nothing

### ISSUE THREAD

**Issue by oraclerob** [upstreamplugin/UpStream#900]
## Expected Behavior

Adding a milestone using the "add milestone" on the front end, or adding via the backend and updating should add the milestone.

## Current Behavior

Since the 1.29.3 update, we cannot add any milestones. The screen returns without the new milestone added.

## Possible Solution

Possible issue with the caching milestone function similar to #899

**Comment by westbourneintl** [comment]
Do you have any other information on this? I can't reproduce it and haven't gotten any emails from anyone else about it, so I doubt it's the caching.  

Is this true for admins too?

**Comment by oraclerob** [comment]
Yes this is happening for admins too. I have reproduced it on 2 separate instances. However my development instance works. I believe this is something to do with the date formating. Here is the screen:

![image](https://user-images.githubusercontent.com/11176397/67854630-9f956d80-fb4b-11e9-887a-509e62dc50a4.png)

Here is the error.log

PHP message: PHP Warning:  count(): Parameter must be an array or an object that implements Countable in /usr/share/nginx/html/prod/wp-content/plugins/upstream-calendar-view/CalendarView/Calendar.php on line 1035
PHP message: PHP Warning:  Invalid argument supplied for foreach() in /usr/share/nginx/html/prod/wp-content/plugins/upstream-calendar-view/CalendarView/Calendar.php on line 1036
PHP message: PHP Warning:  Invalid argument supplied for foreach() in /usr/share/nginx/html/prod/wp-content/plugins/upstream-calendar-view/CalendarView/Calendar.php on line 1194
PHP message: PHP Warning:  count(): Parameter must be an array or an object that implements Counta

My instance that works note how the date format is different?

![image](https://user-images.githubusercontent.com/11176397/67854790-ed11da80-fb4b-11e9-920b-e2dc3daeca84.png)

![image](https://user-images.githubusercontent.com/11176397/67854882-33ffd000-fb4c-11e9-9e5f-61aa4e5d39d2.png)

For sorting purposes we have changed the date format for my client wordpress:

Hope that helps.

**Comment by westbourneintl** [comment]
Hi Rob,

I changed the date formats and time zone and still can't reproduce that problem.

One question: did you place the edit you made [here](https://github.com/upstreamplugin/UpStream-Copy-Project/issues/19) ( https://github.com/upstreamplugin/UpStream-Copy-Project/issues/19 ) on the server? That would explain the warnings.

**Comment by oraclerob** [comment]
This is strange, now I can only reproduce this in the production instance. When I add a milestone you get the success message, however no milestone appears in the list. Any debugging clues you can think of?

![image](https://user-images.githubusercontent.com/11176397/67860075-fbfe8a00-fb57-11e9-8b36-7db9515236f5.png)

**Comment by westbourneintl** [comment]
I'll keep looking at it and get back to you

**Comment by westbourneintl** [comment]
I still can't find any way to reproduce this...any other clues to the differences between the instances?

**Comment by oraclerob** [comment]
The instances are the same, except more recent data in prod than dev. My customer wouldn't be using the milestones at all if the progress % worked without adding a milestone. He adds a milestone so the progress correctly updates, otherwise it's always 0%. I can't remember exactly, but I think this is still an outstanding issue.
